### PR TITLE
Add CreatePokerGameCommand and project structure

### PR DESCRIPTION
**Pull request by GMadorell**

### CODE REVIEW

**Comment by rgomezcasas** [comment]
You're coupled to `Future`

File: src/main/scala/com/lambtors/poker_api/module/poker/application/create/PokerGameCreator.scala [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+import com.lambtors.poker_api.module.poker.domain.PokerGameRepository
+import com.lambtors.poker_api.module.poker.domain.error.PokerGameAlreadyExisting
+import com.lambtors.poker_api.module.poker.domain.model.{AmountOfPlayers, GameId, PokerGame}
+
+final class PokerGameCreator(pokerGameRepository: PokerGameRepository)(implicit ec: ExecutionContext) {
+  def create(amountOfPlayers: AmountOfPlayers, gameId: GameId): Future[Unit] = {
+    pokerGameRepository
+      .search(gameId)
+      .flatMap(_.fold(pokerGameRepository.insert(PokerGame(gameId, amountOfPlayers)))(_ =>
```
Comment (danieljimenezr) [inline src/main/scala/com/lambtors/poker_api/module/poker/application/create/PokerGameCreator.scala]: gerard, fold is the next step, use ifs in our first iteration

File: src/main/scala/com/lambtors/poker_api/module/poker/application/create/PokerGameCreator.scala [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+import scala.concurrent.{ExecutionContext, Future}
+
+import com.lambtors.poker_api.module.poker.domain.PokerGameRepository
+import com.lambtors.poker_api.module.poker.domain.error.PokerGameAlreadyExisting
+import com.lambtors.poker_api.module.poker.domain.model.{AmountOfPlayers, GameId, PokerGame}
+
+final class PokerGameCreator(pokerGameRepository: PokerGameRepository)(implicit ec: ExecutionContext) {
+  def create(amountOfPlayers: AmountOfPlayers, gameId: GameId): Future[Unit] = {
+    pokerGameRepository
+      .search(gameId)
```
Comment (danieljimenezr) [inline src/main/scala/com/lambtors/poker_api/module/poker/application/create/PokerGameCreator.scala]: save search result in a val for more imperative programming in our first step

**Review by danieljimenezr — CHANGES_REQUESTED** [review]

**Comment by GMadorell** [comment]
@rgomezcasas that's the point! we want to show an evolution from purely imperative code to a more functional architecture :)

File: src/main/scala/com/lambtors/poker_api/module/poker/application/create/PokerGameCreator.scala (hunk 2 above)
Comment (GMadorell) [reply src/main/scala/com/lambtors/poker_api/module/poker/application/create/PokerGameCreator.scala]: we still have to flatmap the result, so I think it makes more sense to flatmap directly

**Pull request merged**